src/test/kotlin/uk/gov/dluhc/printapi/client/BankHolidayDataFileClientTest.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.dluhc.printapi.client
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import uk.gov.dluhc.printapi.client.BankHolidayDivision.ENGLAND_AND_WALES
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import java.time.LocalDate
+
+internal class BankHolidayDataFileClientTest : IntegrationTest() {
```

Comment:
Do you really want this to be an integration test?
Integration tests are slow, and all you're really doing is unit testing the `getBankHolidayDates` , based on data that is not in the control of the test (ie. the json file)

Does the design of the client class you are testing allow you to pass in a data file that might be a little closer to the test class?

Reply:
The idea is that each implementation of `BankHolidayDataClient` knows how to retrieve the data it needs. In this case, I want to test that `BankHolidayDataFileClient` reads and deserialises the JSON file correctly, in addition to returning the correct dates. We will have a different test when we start retrieving the bank holiday data from S3 (which we consider using Wiremock for).